src/extension.ts

```diff
@@ -761,11 +761,24 @@ function launchMetals(
           "Test Explorer"
         );
 
+      const getTestEnvVars = () => {
+        const envVars = getValueFromConfig<Record<string, string>>(
+          config,
+          "testEnvironmentVariables",
+          {}
+        );
+        return envVars;
+      };
+
       // vscodeTextExplorer can be undefined e.g. for eclipse theia,
       // see https://github.com/scalameta/metals-vscode/discussions/1244
       const istTestManagerDisabled =
         !vscodeTextExplorer || getTestUI() === "Code Lenses";
-      const testManager = createTestManager(client, istTestManagerDisabled);
+      const testManager = createTestManager(
+        client,
+        istTestManagerDisabled,
+        getTestEnvVars()
```

Comment:
Looks like maybe we should make it lazy actually, since now you need to restart the plugin. Could you take a look? I haven't noticed the issue before